HIRS_AttestationCAPortal/src/main/java/hirs/attestationca/portal/util/CertificateStringMapBuilder.java

```diff
@@ -314,13 +344,35 @@ public static Certificate containsAllChain(
             if (platformConfiguration != null) {
                 //Component Identifier
                 data.put("componentsIdentifier", platformConfiguration.getComponentIdentifier());
+                //Component Identifier URI
+                data.put("componentsIdentifierURI", platformConfiguration
+                        .getComponentIdentifierUri());
                 //Platform Properties
                 data.put("platformProperties", platformConfiguration.getPlatformProperties());
                 //Platform Properties URI
                 data.put("platformPropertiesURI", platformConfiguration.getPlatformPropertiesUri());
             }
             //TBB Security Assertion
             data.put("tbbSecurityAssertion", certificate.getTBBSecurityAssertion());
+
+            if (certificate.getPlatformSerial() != null) {
+                // link certificate chain
+                List<PlatformCredential> chainCertificates = PlatformCredential
+                        .select(certificateManager)
+                        .byBoardSerialNumber(certificate.getPlatformSerial())
+                        .getCertificates().stream().collect(Collectors.toList());
+
+                data.put("numInChain", chainCertificates.size());
+                Collections.sort(chainCertificates, new Comparator<PlatformCredential>() {
+                    @Override
+                    public int compare(final PlatformCredential obj1,
+                            final PlatformCredential obj2) {
+                        return obj1.getBeginValidity().compareTo(obj2.getBeginValidity());
+                    }
+                });
+
+                data.put("chainCredentials", chainCertificates);
```

Comment:
Should we call this key chainCertificates for consistency?

Reply:
Agreed and updated to certificates.